src/actions/channel_categories.ts

```diff
@@ -245,3 +247,86 @@ function removeItem<T>(array: T[], item: T) {
     result.splice(index, 1);
     return result;
 }
+
+(global as any).createCategory = createCategory;
+
+export function createCategory(teamId: string, displayName: string, channelIds: $ID<Channel>[] = []): ActionFunc {
+    return (dispatch: DispatchFunc, getState: GetStateFunc) => {
+        // TODO contact the server to do this
+
+        const newCategory: ChannelCategory = {
+            id: generateId(),
+            team_id: teamId,
+            type: CategoryTypes.CUSTOM,
+            display_name: displayName,
+            sorting: Sorting.NONE,
+            channel_ids: channelIds,
+        };
+
+        const state = getState();
+
+        const categoryIds = getCategoryIdsForTeam(state, teamId);
+        const favoritesIsFirst = categoryIds.length > 0 && getCategory(state, categoryIds[0]).type === CategoryTypes.FAVORITES;
+
+        const newCategoryIds = [...categoryIds];
+
+        // Place the new category relative to other categories
+        if (favoritesIsFirst) {
+            // Place the new category after the favorites category if it comes first
+            newCategoryIds.splice(1, 0, newCategory.id);
+        } else {
+            // Place the new category first
+            newCategoryIds.unshift(newCategory.id);
+        }
+
+        const categoriesToUpdate = [newCategory];
+
+        // Remove the provided channels from any existing categories they may have existed in
+        if (channelIds.length > 0) {
+            for (const categoryId of categoryIds) {
+                const category = getCategory(state, categoryId);
+
+                if (!category.channel_ids.some((channelId) => channelIds.includes(channelId))) {
```

Comment:
3/5 Can we revert this logic for readability, the 'continue' is not necessary since we can push only when the condition is true

Reply:
Yeah, I can do that. At some point, we decided that early returns and early continues were preferable when writing branching logic, but having the condition inverted here does hurt that.